### DISCUSSION TITLE
create procedure interface and prototypes via a form

### DISCUSSION THREAD

**Discussion by lgiammattei** [codefori/vscode-ibmi#1247] [Ideas]
Hi everybody, one feature i've always liked from RDi is the way to create procedure interface and prototype via a guided form like the one i am attaching (my RDi installation is in italian)
![Cattura](https://user-images.githubusercontent.com/54810977/233099977-e1a32869-60c4-44c3-b186-01fb75712477.PNG)
![Cattura01](https://user-images.githubusercontent.com/54810977/233100021-1f5c87dd-4286-4099-92c0-17fae05717c0.PNG)
![Cattura02](https://user-images.githubusercontent.com/54810977/233100036-b2b9fa48-de2b-42f9-8bdd-b2f962fe6fe4.PNG)
![Cattura03](https://user-images.githubusercontent.com/54810977/233100056-ca615b58-1810-4b27-ae9d-3320cc08169e.PNG)
Do you think that it will be possible to create something like this in VSC? I can try to work it on my own and then contribute to the project if you think also other user might be interested in such a feature.
Obiouvsly the resulting code (old p e d code) will be in free format.

**Comment by worksofliam** [comment]
@sebjulliand @chrjorgensen I am not sure how I feel about it.

While it would be cool to put our UI framework to the test, I don't know if this kind of functionality belongs inside of VS Code.

**Reply by lgiammattei** [reply]
@worksofliam why do you think that this kind of functionality doesn't belongs to the VS Code environment?

**Reply by worksofliam** [reply]
@lgiammattei Building UI like this doesn't really fit in with the philosophy of how VS Code wants it to work. We are already building UI that is not consistent with VS Code and that already pushes it.

That's not to say we won't work on this, but generally for a feature like this, it would need a lot of support behind it.

Hope that makes sense!

**Reply by lgiammattei** [reply]
@worksofliam Thanks for your reply. I am new to VS Code, but in the last 20 years I have been using Eclipse, WDSC, and even PHP Storm and NetBeans in a couple of different projects. 
In the last ten years my work has mainly focused on Power I and rpg(le) and RDi has been my daily bread.
Now I know that each of these tool bring also a philosophy of work, but in the end they are tools. 
Whenever they allow me to write better code faster they're good. Otherwise they're bad. Now I'd like to dig deeper into the development philosophy behind VS Code in the meantime I can say that I like VS Code since allow me to have only in one place, a place that run lightly and smooher than other eclipse's based ide, all the things i need for my job, and I really appreciate all the work and efforts you are putting into these projects for building a better evironment for Power I coders. If there will be the chance I'll be happy to give my contribution to your work.
